### PR TITLE
Parallelize department scraping

### PR DESCRIPTION
**Pull request by Connorelsea**
Parallelize department scraping through Bluebird and promise mapping

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 const express = require('express')
-
+const pretty = require("./pretty")
 const scrape = require("./scrape")
 
-scrape()
+const departments = require("./departments")
+//const departments = ["COMPUTER SCIENCE", "HONORS"]
```
Comment (cmwall) [inline index.js:6]: Comment

File: scrape.js [hunk 2]
```diff
@@ -1,44 +1,53 @@
...
-    return
-  }
+const handleRequestResponse = (body) => {
 
   let $ = cheerio.load(body)
-  let pre = $("pre")
+  let pre = $("pre").html()
+
+  if (!pre) {
+    //console.warn("Warning: Error on empty pre, most likely no courses in department")
```
Comment (cmwall) [inline scrape.js:33]: comment

File: scrape.js [hunk 3]
```diff
@@ -134,7 +143,9 @@ const parseLines = lines => {
     }
   }
 
-  console.log(pretty(sections))
+  //console.log("returning sections")
```
Comment (cmwall) [inline scrape.js:146]: comment

**Review by cmwall — APPROVED** [review]
👍

File: scrape.js (hunk 2 above)
Comment (cmwall) [reply scrape.js:33]: also doesn't eslint not like us to use brackets for single line if statements?

Comment (Connorelsea) [reply scrape.js:33]: yeah probably not. Changed it to this

image

**Pull request merged**